tx-pool/src/component/pending.rs

```diff
@@ -228,6 +240,20 @@ mod tests {
             tx1.proposal_short_id(),
         ];
         assert_eq!(txs_sorted_by_fee_rate, expect_result);
+
+        let sorted_fee_and_relation = pool
+            .sorted_fee_and_relation()
+            .iter()
+            .map(|key| key.id.clone())
+            .collect::<Vec<_>>();
+
+        let expect_result = vec![
+            tx1.proposal_short_id(),
+            tx2.proposal_short_id(),
+            tx3.proposal_short_id(),
+            tx4.proposal_short_id(),
```

Comment:
The best `expect_result` is 
```
tx1, tx2, tx4, tx3
```

Because tx4 fee_rate is higher than tx3 and they don't have the dependency relation. Here we make a compromise.

Reply:
No tx4 will includes `tx1, tx2`, it's fee rate better than tx3

Reply:
> No tx4 will includes `tx1, tx2`, it's fee rate better than tx3

Yes. tx4 should come first, as I said before.